src/wallet/spend.cpp

```diff
@@ -1034,7 +1052,7 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     if (nBytes == -1) {
         return util::Error{_("Missing solving data for estimating transaction size")};
     }
-    CAmount fee_needed = coin_selection_params.m_effective_feerate.GetFee(nBytes) + result.GetSummedBumpFees();
+    CAmount fee_needed = coin_selection_params.m_effective_feerate.GetFee(nBytes) + result.GetSummedBumpFees() - result.GetBumpFeeDiscount();
```

Comment:
In 0f6c13665c4ab7d4928ff0fa63c4e755667f7fd6 "Amend bumpfee for inputs with overlapping ancestry"

`GetBumpFeeDiscount` is only used here. It doesn't seem like it's useful to require the caller to calculate the final bumpfee rather than having `SelectionResult` just do that when the the discount is applied and have a single method to return the bumpfees required for a particular selection.

Reply:
I’ve been doing some staring at this today. It seems to me that the waste score of finished input sets should also be adjusted according to the bump fee discount. Gotta do more pondering, will get back to this.

Reply:
Yes, the discount value is a function of the effective fee rate, therefore it has to be taken into account in waste calculation. The problem is that GetSelectionWaste doesn't use SelectionResult.GetEffectiveValue() which already accounts for bump fee discoun

Reply:
Okay, I replaced `GetSummedBumpFees()` with `GetTotalBumpFees()` which deducts the `bump_fee_discount` if it has been set.